domain/src/main/java/com/teamagam/gimelgimel/domain/messages/MessagePresentation.java

```diff
@@ -25,4 +30,41 @@ public boolean isFromSelf() {
     public boolean isShownOnMap() {
         return mIsShownOnMap;
     }
+
+    public boolean isRead() {
+        return mIsRead;
+    }
+
+    static class Builder {
+        private Message mMessage;
+        private boolean mIsFromSelf;
+        private boolean mIsShownOnMap;
+        private boolean mIsRead;
+
+        public Builder(Message message) {
+            mMessage = message;
+        }
+
+        Builder setIsFromSelf(
+                boolean isFromSelf) {
+            mIsFromSelf = isFromSelf;
+            return this;
+        }
+
+        Builder setIsShownOnMap(
+                boolean isShownOnMap) {
+            mIsShownOnMap = isShownOnMap;
+            return this;
+        }
+
+        Builder setIsRead(
+                boolean isRead) {
+            mIsRead = isRead;
+            return this;
+        }
+
+        MessagePresentation build() {
+            return new MessagePresentation(mMessage, mIsFromSelf, mIsShownOnMap, mIsRead);
```

Comment:
I suppose the non-set fields are set by default to false, Is this a desired behavior? In other words, am I allowed to call new Builder.setIsFromSelf(true).build() for example? It will produce an "uncontrolled" MessagePresentation...